e2e/app/sidebar/base-environment-panel.ts

```diff
@@ -0,0 +1,25 @@
+import BaseSidebar from './base-sidebar';
+import { Page } from 'puppeteer';
+import { SideBarLink } from 'app/text-labels';
+import { logger } from 'libs/logger';
+
+// for logic common to environment-related panels (user apps and runtimes)
```

Comment:
we might be able to move this logic into `BaseSidebar` but I was less sure about the other few subclasses.